### ISSUE TITLE
New `raco setup` terminal control not enabled in Alacritty

### ISSUE THREAD

**Issue by benknoble** [racket/racket#4953]
**What version of Racket are you using?**
8.12 [cs]

**What program did you run?**
`raco setup`

**What should have happened?**
I'm expecting the "new" output flavor from #4878 and not seeing it.

**Please include any other relevant details**
- macOS
- alacritty v0.13.1

---

I'm inclined to believe it's Alacritty's fault and not Racket's, but I'd like to understand what the problem is so I can fix it :) I assume this feature depends on some terminal control codes? If so, which? Is there a smaller, hackable test case I can use to see what's going on?

**Comment by benknoble** [comment]
The docs site doesn't list docs for a `#%terminal` module, so I guess that's https://github.com/racket/racket/blob/master/racket/src/cs/io/terminal.ss ?

**Comment by LiberalArtist** [comment]
> The docs site doesn't list docs for a `#%terminal` module, so I guess that's https://github.com/racket/racket/blob/master/racket/src/cs/io/terminal.ss ?

Yes, and equivalent code for BC: it's one of the "primitive modules with names that start with `#%`" mentioned [here](https://docs.racket-lang.org/reference/running-sa.html#(part._init-actions)).

The `foreign-procedure` forms are using the procedures implemented in https://github.com/racket/racket/blob/28157ba88d48fe645563f46f6c00d6626b3428fa/racket/src/ChezScheme/c/expeditor.c .

**Comment by benknoble** [comment]
Hm. It seems that my installation of 8.12 CS does not have the changes from #4878 (I went to copy and play with `collects/setup/private/setup-fprintf.rkt` and it didn't exist; similarly, `collects/setup/setup-core.rkt` still has the old `setup-*printf` functions.)

I must have misunderstood when this feature was being released?

**Comment by benknoble** [comment]
Checking `git log --graph --oneline --first-parent master v8.12` _I think_ indicates those changes weren't pulled into v8.12; `git log v8.12 -- racket/collects/setup/private/setup-fprintf.rkt` is clear they weren't. It was hard to follow the graph until I used `--date-order`, at which point I think I reconstructed that
- the `release` branch was created from 69dd3e9921 (tidy up a little bit, 2024-01-05) with commit 4c0b5e1be8 (Alpha version number for the v8.12 release, 2024-01-07)
- the `stable` branch continued from the previous release (they were merged together in 116a2d92bd (Merge branch 'stable' into release, 2024-02-04))
- master was fast-forwarded onto the PR in question in 251ac7e960 (add a progress bar, 2023-12-25); the commit was authored before the branch in question, but not committed until 2024-01-07  at 16:14-0600 (after the release branch's first commit at 12:54+0000; use `git show -s --format=fuller`).

So now I think it all makes sense 😅

**Comment by rfindler** [comment]
Yes, I think you're right: the changes weren't included in 8.12. I don't see them in my copy of 8.12 anyway. (I'm usually pretty conservative about including stuff in releases and preferring to put stuff on master right after a release branch is created so that fits too.)
